Add AKS platform support (#219)
kinvolk/lokomotive: docs/quickstarts/aks.md

# Lokomotive AKS quickstart guide

## Contents

* [Introduction](#introduction)
* [Requirements](#requirements)
* [Step 1: Install lokoctl](#step-1-install-lokoctl)
* [Step 2: Set up a working directory](#step-2-set-up-a-working-directory)
* [Step 3: Set up Azure credentials from environment variables](#step-3-set-up-zzure-credentials-from-environment-variables)
* [Step 4: Prepare AKS credentials](#step-4-prepare-aks-credentials)
* [Step 5: Define cluster configuration](#step-5-define-cluster-configuration)
* [Step 6: Create Lokomotive cluster](#step-6-create-lokomotive-cluster)
* [Verification](#verification)
* [Cleanup](#cleanup)
* [Conclusion](#conclusion)
* [Next steps](#next-steps)

## Introduction

This quickstart guide walks through the steps needed to create a Lokomotive cluster on AKS.

By the end of this guide, you'll have a production-ready Kubernetes cluster running on Azure AKS.

_Note: Lokomotive on AKS currently provides Kubernetes 1.16 as opposed to other platforms, which provide 1.18. This is because of limitations of Azure platform._

## Requirements

* Basic understanding of Kubernetes concepts.

> * [Azure account.](https://azure.microsoft.com/en-us/free/)

Reviewer: Nit
```suggestion
* [Azure account](https://azure.microsoft.com/en-us/free/).
```

* `kubectl` installed locally to access the Kubernetes cluster.

## Steps

### Step 1: Install lokoctl

lokoctl is a command-line interface for Lokomotive.

To install `lokoctl`, follow the instructions in the [lokoctl installation](../installer/lokoctl.md)
guide.

### Step 2: Set up a working directory

It's better to start fresh in a new working directory, as the state of the cluster is stored in this
directory.

This also makes the cleanup task easier.

```console
mkdir -p lokomotive-infra/myakscluster
cd lokomotive-infra/myakscluster
```

### Step 3: Set up Azure credentials from environment variables

> To create AKS resource in Azure, you need to authenticate to it first. Follow [Authenticating to Azure](https://www.terraform.io/docs/providers/azurerm/index.html#authenticating-to-azure) to set up environment variables required for creating AKS cluster.

Reviewer: ```suggestion
To create an AKS resource in Azure, you need to authenticate to it first. Follow [Authenticating to Azure](https://www.terraform.io/docs/providers/azurerm/index.html#authenticating-to-azure) to set up environment variables required for creating AKS cluster.
```

### Step 4: Prepare AKS credentials

> AKS cluster requires set of service principle credentials to run, as it talks to Azure API to create Load Balancers, Disks and other objects. Depending on your level of privileges in Azure tenant, there are different ways to provide them.

Reviewer: ```suggestion
An AKS cluster requires set of service principal credentials to run, as it talks to Azure API to create Load Balancers, Disks and other objects. Depending on your level of privileges in Azure tenant, there are different ways to provide them.
```

> An AKS cluster requires set of service principal credentials to run, as it talks to Azure API to create Load Balancers, Disks and other objects. Depending on your level of privileges in Azure tenant, there are different ways to provide them.

Reviewer: ```suggestion
An AKS cluster requires a set of service principal credentials to run, as it talks to Azure API to create Load Balancers, Disks and other objects. Depending on your level of privileges in your Azure tenant, there are different ways to provide them.
```

#### Azure AD Application Creator (full automation)

> If you are Azure AD administrator or if your Azure user has permissions to create Azure AD Applications then you don't need to prepare anything manually. When configuring a cluster, set `application_name` property to e.g. cluster name and `lokoctl` will create Azure AD application for you, together with associated service principle and credentials. Those credentials will be automatically used for running AKS.

Reviewer: ```suggestion
If you are an Azure AD administrator or if your Azure user has permissions to create Azure AD Applications then you don't need to prepare anything manually. When configuring a cluster, set the `application_name` property to e.g. the cluster name and `lokoctl` will create Azure AD application for you, together with the associated service principal and credentials. Those credentials will be automatically used for running AKS.
```

> If you are an Azure AD administrator or if your Azure user has permissions to create Azure AD Applications then you don't need to prepare anything manually. When configuring a cluster, set the `application_name` property to e.g. the cluster name and `lokoctl` will create Azure AD application for you, together with the associated service principal and credentials. Those credentials will be automatically used for running AKS.

Reviewer: Nit: line wrapping or at least one sentence per line makes changes easier to review later.

Author: Added some line breaks :)

#### Subscription collaborator

> If you are a user with full administrative access to your subscription, then you need to ask your administrator to create Azure AD application for you and provide you Service Principle Client ID and Client secret, which will be used by AKS cluster.

Reviewer: ```suggestion
If you are a user with full administrative access to your subscription, then you need to ask your administrator to create Azure AD application for you and provide you a Service Principal Client ID and a Client secret, which will be used by AKS cluster.
```

#### Resource group collaborator

> If your Azure user has only access to single Resource Group, you must set `manage_resource_group` property to `false`, as otherwise `lokoctl` will try to create Resource Group for you.

Reviewer: ```suggestion
If your Azure user has only access to a single Resource Group, you must set `manage_resource_group` property to `false`, as otherwise `lokoctl` will try to create Resource Group for you.
```

> If your Azure user has only access to a single Resource Group, you must set `manage_resource_group` property to `false`, as otherwise `lokoctl` will try to create Resource Group for you.

Reviewer: I've seen this pretty often, it's either "you must set `prop` to..." or "you must set the `prop` property to...".

```suggestion
If your Azure user has only access to a single Resource Group, you must set the `manage_resource_group` property to `false`, as otherwise `lokoctl` will try to create Resource Group for you.
```

> You also need Service Principle credentials, as explained in [#subscription-collabolator](#subscription-collabolator).

Reviewer: ```suggestion
You also need Service Principal credentials, as explained in [#subscription-collabolator](#subscription-collabolator).
```
